go/nbs/table_set.go

```diff
@@ -4,64 +4,94 @@
 
 package nbs
 
-import (
-	"bytes"
-	"io"
-	"io/ioutil"
-	"os"
-	"path/filepath"
-	"sort"
-	"sync"
-
-	"github.com/attic-labs/noms/go/d"
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/s3"
-)
-
-const defaultS3PartSize = 5 * 1 << 20 // 5MiB, smallest allowed by S3
-
 func newS3TableSet(s3 s3svc, bucket string) tableSet {
-	return tableSet{p: s3TablePersister{s3, bucket, defaultS3PartSize}}
+	return tableSet{p: s3TablePersister{s3, bucket, defaultS3PartSize}, rl: make(chan struct{}, 5)}
 }
 
 func newFSTableSet(dir string) tableSet {
-	return tableSet{p: fsTablePersister{dir}}
+	return tableSet{p: fsTablePersister{dir}, rl: make(chan struct{}, 5)}
 }
 
 // tableSet is an immutable set of persistable chunkSources.
 type tableSet struct {
 	chunkSources
-	p tablePersister
+	p  tablePersister
+	rl chan struct{}
```

Comment:
I'm ok with this for now, but it feels pretty at odds with the idea that the tableSet being an immutable thing, that it has a rate limiter attached to it.